File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function solution(s) {
    let count = 0;
    const arr = s.split('')
    for(let i=0; i<s.length-1; i+=1){
        if(i>0) changeArr(arr);
        if(isTrue(arr)) count+=1;
    }
    
    return count
}

function changeArr(arr){
    const tmp = arr[0];
    arr.push(tmp);
    arr.shift();
}

function isTrue(input){
    if(input.length%2!==0) return false
    const match = {'(':')','[':']','{':'}'}
    const stack = []
    for(let i=0; i<input.length; i+=1){
        const c = input[i]
        if(c==='('||c==='['||c==='{'){
            stack.push(c)
        }
        
        if(c===')'||c===']'||c==='}'){
            if(stack.length===0) return false;
            
            if(c!==match[stack.pop()]) return false
        }
        
        
    }
    
    return true
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    count = 0
    s = list(x for x in s)
    for i in range(len(s)):
        if correct(s): count += 1
        tmp = s.pop(0)
        s.append(tmp)
    return count

def correct(s):
    dict = { ')':'(', ']':'[', '}':'{' }
    push = [ '(', '[', '{' ]
    stack = []
    for i in s:
        if i in push:
            stack.append(i)
        else:
            if(len(stack)==0): return False
            pop = stack.pop()
            if(dict[i]!=pop):
                return False
    return len(stack)==0